spark/src/main/scala/org/apache/zeppelin/spark/utils/SparkSqlCompleter.scala

```diff
@@ -0,0 +1,38 @@
+package org.apache.zeppelin.spark.utils
+
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.types.{StructType, StructField}
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+class SparkSqlCompleter(val sqlContext: SQLContext) {
+
+  private val keywords = Source.fromInputStream(this.getClass.getResourceAsStream("/sparksql.txt")).getLines().toSet
+
+  def completion(buf: String, cursor: Int): java.util.List[String] = {
+    import sqlContext.implicits._
+
+    val tables = sqlContext.tables().select('tableName).collect().map(_.getString(0))
+    val fields = tables.flatMap(getFields)
```

Comment:
how expansive is it to call collect() or table.schema.field on sqlContext for every call to completion?
is there a way to cache it some how?